### PR TITLE
build: bake aws into the builder image

### PR DESCRIPTION
**Pull request by petermattis**
Release note: None

### CODE REVIEW

**Comment by petermattis** [comment]
Gah, apparently this installs a version of `aws` that is too old:

```
> /home/agent/work/.go/bin/roachprod create teamcity-968270-kv95-encrypt-false-nodes-1 -n 2 --clouds=aws --aws-machine-type-ssd=m5d.2xlarge
Creating cluster teamcity-968270-kv95-encrypt-false-nodes-1 with 2 nodes
2018/10/16 00:06:48 imported ${HOME}/.ssh/id_rsa.pub as teamcity-runner-w3i2u7tBwUAjM4_HewnqvX8mV44= in region us-east-2
2018/10/16 00:06:49 imported ${HOME}/.ssh/id_rsa.pub as teamcity-runner-w3i2u7tBwUAjM4_HewnqvX8mV44= in region us-west-2
2018/10/16 00:06:49 usage: aws [options] <command> <subcommand> [<subcommand> ...] [parameters]
To see help text, you can run:

  aws help
  aws <command> help
  aws <command> <subcommand> help

Unknown options: --tag-specifications, ResourceType=instance,Tags=[{Key=Lifetime,Value=12h0m0s},{Key=Name,Value=teamcity-968270-kv95-encrypt-false-nodes-1-0001},{Key=Roachprod,Value=true},]
```

I saw exactly this same error locally and had to upgrade `aws`. My apt-get fu is weak. How do I track down a newer apt-installable version?

**Comment by petermattis** [comment]
Switched to explicitly download the awslci-bundle from amazon and installing it. Easier than figuring out if there is an up to date apt-get source or using pip.

**Comment by benesch** [comment]
> Switched to explicitly download the awslci-bundle from amazon and installing it. Easier than figuring out if there is an up to date apt-get source or using pip.

That's exactly what I was about to recommend. I'm not seeing a maintained PPA for awscli, and getting pip to work right in the builder would be a hassle, so I think the bundle is the way to go.

**Review by benesch — APPROVED** [review]

**Comment by petermattis** [comment]
Ok, making progress: https://teamcity.cockroachdb.com/viewLog.html?buildId=968422&buildTypeId=Cockroach_Nightlies_RoachtestNightlyAws&tab=buildLog

File: build/builder/Dockerfile [hunk 1]
```diff
@@ -207,7 +208,15 @@ RUN apt-get install -y --no-install-recommends \
...
     yarn \
-    google-chrome-stable
+    google-chrome-stable \
+    unzip
+
+# awscli - roachtests
+# NB: we don't use apt-get because we need an up to date version of awscli
+RUN curl "https://s3.amazonaws.com/aws-cli/awscli-bundle.zip" -o "awscli-bundle.zip" && \
+  unzip awscli-bundle.zip && \
+  ./awscli-bundle/install -i /usr/local/ aws && \
```
[the remark this answers is not in the document]
Comment (petermattis) [reply build/builder/Dockerfile]: Done.

File: build/builder/Dockerfile [hunk 2]
```diff
@@ -207,7 +208,15 @@ RUN apt-get install -y --no-install-recommends \
     openjdk-8-jre \
     openssh-client \
     yarn \
-    google-chrome-stable
+    google-chrome-stable \
+    unzip
+
+# awscli - roachtests
+# NB: we don't use apt-get because we need an up to date version of awscli
+RUN curl "https://s3.amazonaws.com/aws-cli/awscli-bundle.zip" -o "awscli-bundle.zip" && \
```
[the remark this answers is not in the document]
Comment (petermattis) [reply build/builder/Dockerfile]: Done.

**Comment by petermattis** [comment]
And we're running a roachtest on AWS!

**Review by benesch — COMMENTED** [review]
Doesn't look like you actually pushed your latest revision?

**Comment by petermattis** [comment]
Just adding an extra level of difficulty to the review. Pushed.

**Review by benesch — APPROVED** [review]
LGTM!

**Comment by petermattis** [comment]
Thanks, though see #31416. Some aws initialization or something is missing. I think this PR is fine, but I'll hold off on merging it until we get that issue figured out.

**Comment by petermattis** [comment]
bors r=benesch

**Pull request merged**